### ISSUE TITLE
Can't create diary links in Markdown 

### ISSUE THREAD

**Issue by fancsali** [vimwiki/vimwiki#1412]
Prior to submitting a new issue make sure to complete these steps:

- [X] Include the VimWiki settings from your `.vimrc`
```
let g:vimwiki_list = [{'path': '~/vimwiki/', 'syntax': 'markdown', 'ext': '.md', 'auto_diary_index': 1, 'diary_caption_level': 6, 'auto_tags': 1, 'listsyms': ' .X'}]
let g:vimwiki_global_ext = 0
let g:vimwiki_auto_header = 1
let g:vimwiki_emoji_enable = 0
```

- [X] Include the syntax you are using (default / Markdown / MediaWiki) -- Markdown
- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
  - Type a dare `2024-06-15`
  - Exit edit mode
  - Move the cursor over the timestamp and press enter
  - Instead of using the whole timestamp, Vimwiki onl uses the month and day as the label...
  - ... and will create a link to a sibling of the current page indtead of a diary entry.

Some context (at least a line, or even a few lines before/after the one containing the link should be shown.

- [X] Include the output of `:VimwikiShowVersion`.
```
Version: 2024.01.24
Os: Linux
Vim: 901
Branch: HEAD
Revision: fde35bb
Date: 2024-01-24 20:14:13 -0700
```

**Comment by tinmarino** [comment]
Hi @fancsali,

Thank you for the report. I can reproduce, this is a bug. I'll try to have a look.

### Current behaviour

```markdown
2024.01.24  # Press enter at begining of word
[01 24](2024.01.24.md)   # Obtain this
```


### Expected behaviour

```markdown
2024.01.24  # Press enter at beginning of word
[01 24](2024.01.24.md)   # Obtain this
```

**Comment by tinmarino** [comment]
A quick workaround would be to select the full text to use in visual mode

I fixed it in #1420 (already merged), this were my troubleshot notes.

--- 


A quick workaround would be to select the full text to use in visual mode

```bash
# [Command](Command.md): `set iskeyword+=.` does NOT fix it.

# In base.vim:
vimwiki#base#normalize_link

# In markdown_base.vim:
s:normalize_link_syntax_n

# In vars.vim:
let g:vimwiki_global_vars.rxWord = '[^[:blank:]!"$%&''()*+,:;<=>?\[\]\\^`{}]*[^[:blank:]!"$%&''()*+.,:;<=>?\[\]\\^`{}]'

# In base.vim:
function! vimwiki#base#normalize_link_helper(str, rxUrl, rxDesc, template) abort


# Bug in markdown_base.vim
s:normalize_link_syntax_n()

# Bad argument passing in call:
# Note the ''
let sub = vimwiki#base#normalize_link_helper(lnk,
      \ vimwiki#vars#get_global('rxWord'), '',
      \ vimwiki#vars#get_syntaxlocal('Link1'))
            

# Declaration:
function! vimwiki#base#normalize_link_helper(str, rxUrl, rxDesc, template) abort

# So needed to pass the rxDesc otherwise the URL was converted back to a desc.
```

**Comment by tinmarino** [comment]
@fancsali can you please:

1. Refresh the dev branch, after #1413
2. Check if the issue is fixed
3. Notify here in comment and close the issue if fixed

Once again thank you very much for the report with detailed "steps to reproduce the issue".

**Comment by fancsali** [comment]
Just for the record, in my case dates look like this: `2024-06-23` - nevertheless I can confirm, checking out the `dev` branch at `705ad1e` resolves the issue.
